modules/api/src/main/scala/vinyldns/api/domain/zone/ZoneConnectionValidator.scala

```
@@ -71,6 +71,41 @@ class ZoneConnectionValidator(defaultConnection: ZoneConnection, scheduler: Sche
     }
   }.toResult
 
+  def testDdnsConnectivity(dnsConnection: DnsConnection, zone: Zone): Result[Unit] = {
+    val rs = RecordSet(
+      zone.id,
+      "vinyldns-test",
+      RecordType.TXT,
+      30,
+      RecordSetStatus.Pending,
+      DateTime.now,
+      records = List(TXTData("connection test")))
+
+    def performDdnsChecks(rsList: List[RecordSet]): Result[Unit] =
+      if (rsList.isEmpty) {
+        // Perform add then delete of test record
```

I think you should change the check up.

The logic currently follows:
1. If the test record does not exist, then add it and then delete it
2. If the test record does exist, then delete it and then add it

You will end up in two different places.   If the record does not exist (first case), it will not exist after the logic.   If the record does exist (second case), it will still exist after the logic.

Rather than doing the resolve, I would recommend:

1. Attempt to delete the record.  If you get a not found error coming back, that is ok!  Keep moving, swallowing that error
2. Create the record

I think doing an immediate delete is not what we want, but open to discussion.  DNS servers maybe eventually consistent, so that churn may wind up with some unexpected errors?  Perhaps just leave it there after step 2. 

**Note: we do not need to verify the record was created.  If we get back a NoError on the add we should be good**.

---

@pauljamescleary Thanks for the comment - discussion would definitely be useful for me.
- One caveat that I found without using `resolve` is that if we are deleting a record set that matches the same name and type, we would potentially be re-adding it with different record data, TTL and other settings. I was playing around with `resolve` and I think that `NoError` is also returned in the DNS response for a delete of a non-existent record.
- It might also be confusing if we never delete the record since it would show up in the zone unless we are somehow filtering that out.

With that said, if we are fine with saying that `vinyldns-test` is off-limits to users and will be created in a zone upon connection, then I think a simple delete and add like you suggest would suffice.